ppx_test/test.ml

```diff
@@ -220,6 +220,11 @@ let test_pvar_inherit () =
 let test_pvar_inherit2 () =
   check_marshal_unmarshal (`four "apple", Rpc.Enum [Rpc.String "four"; Rpc.String "apple"], rpc_of_test_pvar_inherit, test_pvar_inherit_of_rpc)
 
+type enum = [ `x | `y | `z | `default ] [@default `default] [@@deriving rpc]
+type enum_string_map = (enum * string) list [@@deriving rpc]
+let test_enum_string_map () =
+  check_marshal_unmarshal ([`x, "x"; `y, "y"; `z, "z"], Rpc.Dict ["x", Rpc.String "x"; "y", Rpc.String "y"; "z", Rpc.String "z"], rpc_of_enum_string_map, enum_string_map_of_rpc)
+
```

Comment:
Should we also test the use of ` ``default ` here?

Reply:
Ah yes, only had a test for the issue I was trying to fix, not the thing I added to make it work :-)